### PR TITLE
Feature/csi 4973 pvc controller remove pvc from vg

### PR DESCRIPTION
**Pull request by matancarmeli7**
PRs that relates to it:

- #298
-  #299

### CODE REVIEW

File: volume-group-operator/controllers/utils/volumegroup.go [hunk 1]
```diff
@@ -46,3 +67,67 @@ func UpdateVolumeGroupStatusError(client client.Client, instance *volumegroupv1.
 
 	return nil
 }
+
+func GetVGList(logger logr.Logger, client client.Client) (volumegroupv1.VolumeGroupList, error) {
+	logger.Info(messages.ListVolumeGroups)
+	vg := &volumegroupv1.VolumeGroupList{}
+	err := client.List(context.TODO(), vg)
```
Comment (ArbelNathan) [inline volume-group-operator/controllers/utils/volumegroup.go:74]: is there a way to filter here by PVC label and volume group selector?

File: volume-group-operator/controllers/utils/utils.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+limitations under the License.
+*/
+
+package utils
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+func removeByIndexFromPersistentVolumeClaimList(pvcList []corev1.PersistentVolumeClaim,
```
Comment (ArbelNathan) [inline volume-group-operator/controllers/utils/utils.go:23]: sometimes the name with PVC, and others with PersistentVolumeClaim, can we unify to PVC? (same for PV)
```suggestion
func removeByIndexFromPVCList(pvcList []corev1.PersistentVolumeClaim,
```

File: volume-group-operator/controllers/utils/volumegroup.go [hunk 3]
```diff
@@ -46,3 +67,67 @@ func UpdateVolumeGroupStatusError(client client.Client, instance *volumegroupv1.
...
+		logger.Error(err, fmt.Sprintf(messages.FailedToRemovePersistentVolumeClaimFromVolumeGroup,
+			pvc.Namespace, pvc.Name, vg.Namespace, vg.Name))
+		return err
+	}
+	logger.Info(fmt.Sprintf(messages.RemovedPersistentVolumeClaimFromVolumeGroup,
+		pvc.Namespace, pvc.Name, vg.Namespace, vg.Name))
+	return nil
+}
+
+func removePersistentVolumeClaimFromVolumeGroupPVCList(pvc *corev1.PersistentVolumeClaim,
```
Comment (ArbelNathan) [inline volume-group-operator/controllers/utils/volumegroup.go:124]: we can understand what is removed by args
```suggestion
func removeFromVolumeGroupPVCList(pvc *corev1.PersistentVolumeClaim,
```

File: volume-group-operator/controllers/utils/volumegroup.go (hunk 1 above)
Comment (matancarmeli7) [reply volume-group-operator/controllers/utils/volumegroup.go:74]: I didn't find a way, you usually filter objects with labelSelector and the PVC doesn't have labelSelector, so I didn't find a way to get only the VGs that match the PVC in one command, I can open a ticket for after the GA to investigate it, but I didn't find a way when I did it

File: volume-group-operator/controllers/utils/volumegroup.go [hunk 4]
```diff
@@ -46,3 +67,67 @@ func UpdateVolumeGroupStatusError(client client.Client, instance *volumegroupv1.
...
+	pvcListInVG []corev1.PersistentVolumeClaim) []corev1.PersistentVolumeClaim {
```
Comment (ArbelNathan) [inline volume-group-operator/controllers/utils/volumegroup.go:125]: in this context, I don't think it's important to know the VG
```suggestion
	pvcList []corev1.PersistentVolumeClaim) []corev1.PersistentVolumeClaim {
```

File: volume-group-operator/controllers/utils/volumegroupcontent.go [hunk 5]
```diff
@@ -101,3 +106,29 @@ func generateVolumeGroupContentSource(vgcObj *volumegroupv1.VolumeGroupClass, re
...
+		logger.Error(err, fmt.Sprintf(messages.FailedToRemovePersistentVolumeFromVolumeGroupContent,
+			pv.Namespace, pv.Name, vgc.Namespace, vgc.Name))
+		return err
+	}
+	logger.Info(fmt.Sprintf(messages.RemovedPersistentVolumeFromVolumeGroupContent,
+		pv.Namespace, pv.Name, vgc.Namespace, vgc.Name))
+	return nil
+}
+
+func removePersistentVolumeFromVolumeGroupContentPVList(pv *corev1.PersistentVolume,
```
Comment (ArbelNathan) [inline volume-group-operator/controllers/utils/volumegroupcontent.go:125]:
```suggestion
func removeFromPVList(pv *corev1.PersistentVolume,
```

File: volume-group-operator/main.go [hunk 6]
```diff
@@ -31,6 +31,7 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+
```
Comment (ArbelNathan) [inline volume-group-operator/main.go:34]: ?

File: volume-group-operator/controllers/persistentvolumeclaim/persistentvolumeclaim_controller.go [hunk 7]
```diff
@@ -19,6 +19,8 @@ package persistentvolumeclaim
 import (
 	"context"
 
+	csiv1 "github.com/IBM/volume-group-operator/api/v1"
```
Comment (ArbelNathan) [inline volume-group-operator/controllers/persistentvolumeclaim/persistentvolumeclaim_controller.go:22]: please change `PersistentVolumeClaimWatcher` to `PersistentVolumeClaimReconciler`

File: volume-group-operator/main.go (hunk 6 above)
Comment (matancarmeli7) [reply volume-group-operator/main.go:34]: it's my IDE formating

Comment (matancarmeli7) [reply volume-group-operator/main.go]: removed the extra space

File: volume-group-operator/controllers/utils/utils.go (hunk 2 above)
Comment (matancarmeli7) [reply volume-group-operator/controllers/utils/utils.go:23]: done

File: volume-group-operator/controllers/utils/volumegroup.go (hunk 3 above)
Comment (matancarmeli7) [reply volume-group-operator/controllers/utils/volumegroup.go:124]: done

File: volume-group-operator/controllers/utils/volumegroup.go (hunk 4 above)
Comment (matancarmeli7) [reply volume-group-operator/controllers/utils/volumegroup.go:125]: done

File: volume-group-operator/controllers/utils/volumegroupcontent.go (hunk 5 above)
Comment (matancarmeli7) [reply volume-group-operator/controllers/utils/volumegroupcontent.go:125]: done

File: volume-group-operator/controllers/utils/utils.go (hunk 2 above)
Comment (matancarmeli7) [reply volume-group-operator/controllers/utils/utils.go]: and created CSI-5156 ticket

File: volume-group-operator/controllers/persistentvolumeclaim/persistentvolumeclaim_controller.go (hunk 7 above)
Comment (matancarmeli7) [reply volume-group-operator/controllers/persistentvolumeclaim/persistentvolumeclaim_controller.go:22]: [done](https://github.com/IBM/ibm-block-csi-operator/pull/305)

**Review by ArbelNathan — APPROVED** [review]
LGTM

**Pull request merged**